# Assignment 4 - Task 2: Visualizing fire scars through false color
## Section 2

### Author: Josephine Cardelle 

### GitHub repository:

https://github.com/jocardelle/eds220-hwk4

## Load in Libraries

In [1]:
# Load libraries
import os              
import numpy as np
import pandas as pd
import geopandas as gpd

## Read in the data

In [3]:
# Read in the data
fp = os.path.join('data','California_Fire_Perimeters','California_Fire_Perimeters_(all).shp')
fire_perimeter = gpd.read_file(fp)

## Fire perimeter data retrieval and selection

a. Explore the data and write a brief summary of the information you obtained

In [4]:
# Find CRS, if its projected or graphic, ellipsoid, and datum of df
print(f"{'The CRS is:':<25} {fire_perimeter.crs}")
print(f"{'Is the CRS geographic?:':<25} {fire_perimeter.crs.is_geographic}")
print(f"{'Is the CRS projected?:':<25} {fire_perimeter.crs.is_projected}")
print(f"{'ellipsoid:':<25} {fire_perimeter.crs.ellipsoid}")
print(f"{'datum:':<25} {fire_perimeter.crs.datum}")

The CRS is:               EPSG:3857
Is the CRS geographic?:   False
Is the CRS projected?:    True
ellipsoid:                WGS 84
datum:                    World Geodetic System 1984 ensemble


In [5]:
# View the first three rows of the df
fire_perimeter.head(3)

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
0   2023    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17  2023-06-17   
1   2023    CA    LRA     BTU     KAISER  00010225  2023-06-02  2023-06-02   
2   2023    CA    CDF     AEU    JACKSON  00017640  2023-07-01  2023-07-02   

   CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0      5         1          1    5.72913     None       None   
1      5         1          1   13.60240     None       None   
2      2         1          1   27.81450     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None     2020   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None     2020   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None     2020   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...

In [6]:
# Find the size of the df
fire_perimeter.shape

(22261, 19)

In [7]:
# Find what years are included in the df
fire_perimeter['YEAR_'].unique()

array([2023, 2022, 2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014, 2013,
       2012, 2011, 2010, 2009, 2008, 2007, 2006, 2005, 2004, 2003, 2002,
       2001, 2000, 1999, 1998, 1997, 1996, 1995, 1994, 1993, 1992, 1991,
       1990, 1989, 1988, 1987, 1986, 1985, 1984, 1983, 1982, 1981, 1980,
       1979, 1978, 1977, 1976, 1975, 1974, 1973, 1972, 1971, 1970, 1969,
       1968, 1967, 1966, 1965, 1964, 1963, 1962, 1961, 1960, 1959, 1958,
       1957, 1956, 1955, 1954, 1953, 1952, 1951, 1950, 1949, 1948, 1947,
       1946, 1945, 1944, 1943, 1942, 1941, 1940, 1939, 1938, 1937, 1936,
       1935, 1934, 1933, 1932, 1931, 1930, 1929, 1928, 1927, 1926, 1925,
       1924, 1923, 1922, 1921, 1920, 1919, 1918, 1917, 1916, 1915, 1914,
       1913, 1912, 1911, 1910, 1909, 1908, 1907, 1906, 1905, 1903, 1902,
       1900, 1898, 1896, 1895, 1878,    0])

In [8]:
# Find what agencies are included in the df
fire_perimeter['AGENCY'].unique()

array(['CDF', 'LRA', 'CCO', 'FWS', 'USF', 'NPS', 'BLM', 'BIA', 'DOD',
       None, 'OTH'], dtype=object)

Through the initial data exploration, I found the CRS is EPSG 3857. I also found the data is projected and not geographic and the ellipsoid is WGS 84. I viewed the first three columns of the dataframe and was able to look at the different column names. I then looked at the shape of the df and found there are 22261 rows and 19 columns. I also looked at the years that this df covers and the yeaers span from 1878-2023 with a year 0 as well. The agencies included in this df are 'CDF', 'LRA', 'CCO', 'FWS', 'USF', 'NPS', 'BLM', 'BIA', 'DOD', 'OTH', and none.

b. Select the Thomas Fire Boundary that occured in 2017

In [9]:
# Select the Thomas Fire in 2017
thomas_fire = fire_perimeter[(fire_perimeter['FIRE_NAME'] == "THOMAS") & (fire_perimeter['YEAR_'] == 2017)] 
thomas_fire.head()


YEAR_ STATE AGENCY UNIT_ID FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
2654   2017    CA    USF     VNC    THOMAS  00003583  2017-12-04  2018-01-12   

      CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES                    COMMENTS  \
2654      9         7          1   281791.0  CONT_DATE based on Inciweb   

     COMPLEX_NA IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
2654       None    None     None       None     2010   

                                               geometry  
2654  MULTIPOLYGON (((-13316089.016 4088553.040, -13...

c. Save only the Thomas Fire boundary as a geospatial file in the data folder

In [13]:
# Save Thomas Fire as a nc file
thomas_fire.to_file("thomas_fire.geojson", driver="GeoJSON")

d. Explain your reasoning for selecting the geospatial file format